CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required( VERSION 3.0 )
+
+project( cubature )
+
+add_library( cubature SHARED 
+    hcubature.c
+    pcubature.c)
+
+add_executable( htest test.c )
+target_link_libraries( htest cubature m )
+
+add_executable( ptest test.c )
+target_link_libraries( ptest cubature m )
+target_compile_definitions( ptest PRIVATE "-DPCUBATURE" )
```

Comment:
The advantage of `PCUBATURE=1` is that `cmake` will generate the appropriate compiler flag on non-Unixy compilers.